File: BE/living-server/src/main/java/com/living/hana/entity/Building.java
package com.living.hana.entity;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Building {
    
    private Long id;
    private String name;
    private String address;
    private String addressDetail;
    private String zipCode;
    private String buildingType; // OFFICETEL, APARTMENT
    private Integer totalFloors;
    private Integer totalUnits;
    private String status; // ACTIVE, INACTIVE
    private String city; // 시/도 (서울시, 부산시 등)
    private String district; // 구/군 (강남구, 동작구 등)
    private Double latitude; // 위도
    private Double longitude; // 경도
    private String images; // GCP Cloud Storage URL 배열 (JSON 문자열)
    private String createdAt;
    private String updatedAt;
}

File: FE/living-erp/src/types/building.ts
// 건물 관련 타입 정의
export interface Building {
  id: number;
  name: string;
  address: string;
  addressDetail?: string;
  zipCode?: string;
  buildingType: string;
  totalFloors: number;
  totalUnits: number;
  status: string;
  images: string[];
  latitude?: number;
  longitude?: number;
  createdAt: string;
  updatedAt: string;
}

export interface Unit {
  id: number;
  buildingId: number;
  unitNumber: string;
  floor: number;
  type: string;
  size: number;
  monthlyRent: number;
  deposit: number;
  status: 'AVAILABLE' | 'OCCUPIED' | 'MAINTENANCE';
  createdAt: string;
  updatedAt: string;
  building?: Building; // 선택적 건물 정보
}

// API 응답 타입
export interface GetBuildingsResponse {
  success: boolean;
  data: Building[];
  message: string;
}

export interface GetUnitsResponse {
  success: boolean;
  data: Unit[];
  message: string;
}
